File: Array/Product of Array expect Self/By iterative.java
// Method - Iteration(Brute Force)
// Time Complexity T - O(n^2)
// Space Complexity S - O(1)

class Solution {
    public int[] productExceptSelf(int[] nums) {
        
        // inisalize the result
        int[] prod = new int[nums.length];
        
        // outter loop runs for o to n-1 
        for (int i = 0; i< nums.length; i++){

            int product = 1;
            
            // inner loop runs for o to n-1
            for (int j = 0; j < nums.length; j++){
                
                // if i is not j then make a product of nums[j]
                if(i != j){
                    product = product * nums[j];
                }
            }
            
            // assign the product value to the prod[i]th element
            prod[i] = product;
        }
        
        return prod;
    }
}

File: Array/Subarray Product Less than K/By iterative.cpp
// Method - Iterative(BruteForce)
// Time Complexity T - O(n^2), since nested loop takes n^2 time
// Space complexity - O(1), since brute force don't take extra space

class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
         // count variable for counting the no. of subarrays
        int count = 0;
        
        int i, j, product;
     
        // outter loop runs for 0 to n-1 times 
        for (i = 0; i < nums.size(); i++) {
            
            // Counting for single element,
            if (nums[i] < k){
                count++;
            }
                
            // assign the "i" value to the product
            product = nums[i];
            
            // inner loop runs for "i+1" to n-1 times
            for (j = i + 1; j < nums.size(); j++) {
                
                // prosuct of subarray "i" * "j" element
                product = product * nums[j];
                
                // If this product is less
                // than k, then increment the count
                if (product < k){
                    count++;
                }
                // else break the inner loop
                else
                    break;
            }
        }
       return count;  
    }
};
